lib/algo/manifold_interp/SnapshotInterpolator.h

```diff
@@ -0,0 +1,51 @@
+#ifndef included_SnapshotInterpolator_h
+#define included_SnapshotInterpolator_h
+
+#include "Interpolator.h"
+#include <vector>
+#include <string>
+
+namespace CAROM {
+
+class Matrix;
+class Vector;
+
+/**
+ * Implements PCHIP algorithm.  Based on "A METHOD FOR CONSTRUCTING LOCAL MONOTONE
+ * PIECEWISE CUBIC INTERPOLANTS", Fritchs and Butland (1984).  as well as "MONOTONE
+ * PIECEWISE CUBIC INTERPOLATION," Fritsch and Carlson (1980)
+ *
+ */
+class SnapshotInterpolator
+{
+public:
+    SnapshotInterpolator();
+    ~SnapshotInterpolator();
+
+    std::vector<Vector*> interpolate(std::vector<Vector*> snapshot_ts,
+                                     std::vector<Vector*> snapshots,
```

Comment:
`vector` arguments should be passed by reference.

Reply:
Fixed.